### PR TITLE
fix: klaviyo error handling

### PR DESCRIPTION
**Pull request by Gauravudia**
## Description of the change

> Resolved INT-340
Return retryable error if profile Id/ duplicate profile id is not returned in the response from Klaviyo

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/klaviyo/util.js [hunk 1]
```diff
@@ -47,7 +47,9 @@ const getIdFromNewOrExistingProfile = async (endpoint, payload, requestOptions)
     profileId = resp.response?.data?.id;
   } else if (resp.status === 409) {
     const { errors } = resp.response;
-    profileId = errors?.[0]?.meta?.duplicate_profile_id;
+    if (errors && Array.isArray(errors)) {
```
Comment (Gauravudia) [inline src/v0/destinations/klaviyo/util.js:50]: @ujjwal-ab This check is redundant since we are already using optional chaining below.

**Review by koladilip — DISMISSED** [review]

File: src/v0/destinations/klaviyo/util.js (hunk 1 above)
Comment (ujjwal-ab) [reply src/v0/destinations/klaviyo/util.js:50]: We got the bugsnag alert due to this only, as the errors was not present in the `resp.response`. This is for avoiding that.

Comment (ujjwal-ab) [reply src/v0/destinations/klaviyo/util.js]: Got it.Removed.

File: src/v0/destinations/klaviyo/util.js [hunk 2]
```diff
@@ -47,18 +47,27 @@ const getIdFromNewOrExistingProfile = async (endpoint, payload, requestOptions)
...
-    );
+    profileId = errors?.[0]?.meta?.duplicate_profile_id;
+  }
+
+  if (profileId) {
+    return profileId;
+  }
+
+  let statusCode = resp.status;
+  if (resp.status === 201 || resp.status === 409) {
```
Comment (utsabc) [inline src/v0/destinations/klaviyo/util.js:58]: If status is 201 or 409 why are we retrying?

Comment (ujjwal-ab) [reply src/v0/destinations/klaviyo/util.js:58]: If we are getting 409, but in response not getting the profileId. We are retrying in that case and those events get successful after retry.

Comment (Gauravudia) [reply src/v0/destinations/klaviyo/util.js:58]: Yes, for 409 it is getting is succeeded in retry

**Review by utsabc — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Review by anantjain45823 — APPROVED** [review]

**Pull request merged**